otter/auth.py

```diff
@@ -509,6 +601,17 @@ def generate_authenticator(reactor, config):
     """
     # FIXME: Pick an arbitrary cache ttl value based on absolutely no science.
     cache_ttl = config.get('cache_ttl', 300)
+    if config['strategy'] == 'single_tenant':
```

Comment:
For backwards compatibility reasons, it'd be a good idea to change this to `config.get('strategy', 'impersonation')`.